File: plugins/dotnet/robocode.dotnet.robotscs/src/tested/robotscs/GunHeat.cs
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
using Robocode;

namespace tested.robotscs
{
    public class GunHeat : AdvancedRobot
    {
        public override void Run()
        {
            while (true)
            {
                SetFireBullet(3.0);
                Out.WriteLine(Time + " gunHeat after fire: " + GunHeat);
                Execute();
            }
        }
    }
}

File: robocode.tests.robots/src/main/java/tested/robots/GunHeat.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package tested.robots;


import robocode.AdvancedRobot;


/**
 * @author Pavel Savara (original)
 * From bug 2405844
 */
public class GunHeat extends AdvancedRobot {

	public void run() {
		while (true) {
			setFireBullet(3.0);
			out.println(getTime() + " gunHeat after fire: " + getGunHeat());
			execute();
		}
	}
}
